module_10/002_typical_errors.php

```diff
@@ -0,0 +1,58 @@
+<?php
+// Исправьте типовые ошибки
+
+// 1. Использование констант вместо строк
+define('TODAY_IS_MONDAY', false);
+
+// 2. Указание констант вместо ключей в ассоциативных массивах 
+$data = [
+    'FIRST_KEY' => 1,
+];
+echo $data['FIRST_KEY'];
+
+// 3. Использование переменных, которые еще не объявлены
+$x = rand(1, 4);
+$a = 1; // Сделал глобальной переменную $a
+if ($x > 2) {
+    $a = 0;
+}
+echo $a;
+
+// 4. Переопределение значений в суперглобальных переменных
+if ($x > 2) {
+    $_GET['show_info'] = 'y';
```

Comment:
You should avoid redefining of super global variables because it can cause unexpected errors or behaviour of your functions

Reply:
Can i remove excess condition?

if ($x > 2) {
    echo 'done';
}

Reply:
I think it right solution